File: web/shared/src/services/login-service.ts
export class LoginService {
  constructor(api: string) {
    this.api = api;
  }

  api: string;

  private get headers(): HeadersInit {
    return {
      'content-type': 'application/json'
    };
  }

  async login(password: string): Promise<Response> {
    return await fetch(this.api, {
      method: 'POST',
      cache: 'no-cache',
      headers: { ...this.headers, authorization: password }
    });
  }
}

File: apple/TRssReader/Services/LoginService.swift
//
//  LoginService.swift
//  TRssReader
//
//  Created by Ty Hopp on 6/4/23.
//

import Foundation

class LoginService {
    @Sendable func login(password: String) async throws -> (Data, URLResponse) {
        guard let url = URL(string: Env.LOGIN_API) else {
            throw ServiceError.url
        }
        
        var request = URLRequest(url: url, cachePolicy: .reloadIgnoringLocalCacheData)
        
        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
        request.setValue(password, forHTTPHeaderField: "Authorization")
        request.httpMethod = "POST"
        
        return try await URLSession.shared.data(for: request)
    }
}
